libsawtooth/src/client/mod.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright 2020 Bitwise IO
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A trait for implementing a sawtooth client.
+
+mod error;
+pub mod rest;
+
+pub use error::SawtoothClientError;
+
+/// A trait that can be used to interact with a sawtooth node.
+trait SawtoothClient {
+    fn list_batches(&self) -> Result<Box<dyn Iterator<Item = Batch>>, SawtoothClientError>;
+}
+
+/// A struct that represents a batch.
+pub struct Batch {
```

Comment:
This will presumable expand when `get_batch` is implemented, but that can wait for a future PR.